### ISSUE TITLE
vue/require-default-prop not picking up property name

### ISSUE THREAD

**Issue by MathieuDerelle** [vuejs/eslint-plugin-vue#776]
**Tell us about your environment**	
* **ESLint version:**	5.12.0
* **eslint-plugin-vue version:** 5.1.0
* **Node version:**	8.11.4

**Please show your full configuration:**

```javascript
module.exports = {
  root: true,
  env: {
    browser: true,
    node: true
  },
  parserOptions: {
    parser: 'babel-eslint'
  },
  extends: [
    'standard',
    'plugin:vue/recommended'
  ],
  // required to lint *.vue files
  plugins: [
    'vue'
  ],
  // add your custom rules here
  rules: {
    'no-console':  process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',

    "no-multi-spaces": "off",
    "key-spacing": "off",
    "comma-dangle": ["warn", "always-multiline"],

    "vue/max-attributes-per-line": ["warn", {
      "singleline": 3,
      "multiline": {
        "max": 1,
        "allowFirstLine": false
      }
    }],
  },
  globals: {
    CORE_API_BASE_URL: false,
  },
}
```

**What did you do?**	

```vue
<script>
export default {
	props: {
	    tallStyle: Object,
	    'inverse': Boolean,
	    'actions': Array,
	}
}
</script>
```

**What did you expect to happen?**

The warnings should show the property names "tallStyle" and "actions"

**What actually happened?**

```
  95:5   warning  Prop 'tallStyle' requires default value to be set  vue/require-default-prop
  97:5   warning  Prop 'undefined' requires default value to be set  vue/require-default-prop
```

**Comment by ota-meshi** [comment]
I think that the problem will be solved if you change as follows.

```vue
<script>
export default {
	props: {
	    tallStyle: {
	      type: Object,
	      default() {
	        return undefined // The default value you want to set.
	      }
	    },
	    'inverse': Boolean,
	    'actions': {
	      type: Array,
	      default() {
	        return [] // The default value you want to set.
	      }
	    }
	}
}
</script>
```

https://vuejs.org/v2/guide/components-props.html#Prop-Validation

**Comment by armano2** [comment]
@ota-meshi issue is with resolving name of property, right now its `undefined`

in https://github.com/vuejs/eslint-plugin-vue/blob/master/lib/rules/require-default-prop.js#L148

instead of
```js
propName: prop.key.name
```
we should do
```js
propName: utils.getStaticPropertyName(prop.key)
```

**Comment by ota-meshi** [comment]
@armano2 Thank you for pointing that out.
I misunderstood.

I fixed a similar problem a little while ago, but I think that it is correct not to specify the `key` as below.

```js
propName: utils.getStaticPropertyName(prop)
```

I will fix this.
